### PR TITLE
Move and zoom camera to make points visible

### PR DESCRIPTION
**Pull request by sbergmair**
Camera automatically zooms and moves to make given list of points visible.

Issue: https://github.com/fluttercommunity/arcgis_map_sdk/issues/55

### CODE REVIEW

File: arcgis_map_sdk_android/android/src/main/kotlin/dev/fluttercommunity/arcgis_map_sdk_android/ArcgisMapView.kt [hunk 1]
```diff
@@ -269,6 +274,23 @@ internal class ArcgisMapView(
...
+        val future = if(padding != null) mapView.setViewpointGeometryAsync(polyline.extent, padding) else mapView.setViewpointGeometryAsync(polyline.extent)
+        future.addDoneListener {
+            try {
+                result.success(future.get())
+            } catch (e: Exception) {
+                result.error("Error", e.message, e)
+            }
+        }
+    }
+
```
Comment (JulianBissekkou) [inline arcgis_map_sdk_android/android/src/main/kotlin/dev/fluttercommunity/arcgis_map_sdk_android/ArcgisMapView.kt:293]: nit: can you reformat this file?

File: arcgis_map_sdk_ios/ios/Classes/ArcgisMapView.swift [hunk 2]
```diff
@@ -190,6 +191,24 @@ class ArcgisMapView: NSObject, FlutterPlatformView {
             result(success)
         }
     }
+    
+    private func onMoveCameraToPoints(_ call: FlutterMethodCall, _ result: @escaping FlutterResult) {
+        let dict = call.arguments as! Dictionary<String, Any>
+    
+        let payload: MoveToPointsPayload = try! JsonUtil.objectOfJson(dict)
+        let polyline = AGSPolyline(points: payload.points.map { latLng in AGSPoint(x: latLng.longitude,y:latLng.latitude,spatialReference:.wgs84())
```
Comment (JulianBissekkou) [inline arcgis_map_sdk_ios/ios/Classes/ArcgisMapView.swift:199]: I quickly formatted this for you
```suggestion
        let polyline = AGSPolyline(points: payload.points.map { latLng in AGSPoint(x: latLng.longitude, y:latLng.latitude, spatialReference: .wgs84())
```

File: arcgis_map_sdk_ios/ios/Classes/ArcgisMapView.swift [hunk 3]
```diff
@@ -448,3 +467,8 @@ extension AGSBasemapStyle {
         }
     }
 }
+
+struct MoveToPointsPayload : Codable {
+    let points : [LatLng]
+    let padding : Double?
+}
```
Comment (JulianBissekkou) [inline arcgis_map_sdk_ios/ios/Classes/ArcgisMapView.swift:474]:
```suggestion
struct MoveToPointsPayload : Codable {
    let points: [LatLng]
    let padding: Double?
}
```

File: arcgis_map_sdk_method_channel/lib/src/method_channel_arcgis_map_plugin.dart [hunk 4]
```diff
@@ -137,6 +137,21 @@ class MethodChannelArcgisMapPlugin extends ArcgisMapPlatform {
...
+  }) {
+    return _methodChannelBuilder(mapId).invokeMethod<bool>(
+      "move_camera_to_points",
+      {
+        "points": points.map((p) => p.toMap()).toList(),
+        "padding": padding,
+      },
+    ).then((value) => value!);
+  }
+
```
Comment (JulianBissekkou) [inline arcgis_map_sdk_method_channel/lib/src/method_channel_arcgis_map_plugin.dart:154]: No need to force unwrap this value
```suggestion
  @override
  Future<void> moveCameraToPoints({
    required List<LatLng> points,
    required int mapId,
    double? padding,
  }) {
    return _methodChannelBuilder(mapId).invokeMethod<bool>(
      "move_camera_to_points",
      {
        "points": points.map((p) => p.toMap()).toList(),
        "padding": padding,
      },
    );
  }

```

File: example/lib/main.dart [hunk 5]
```diff
@@ -10,8 +10,7 @@ import 'package:flutter/material.dart';
 void main() => runApp(const ExampleApp());
 
 const arcGisApiKey = String.fromEnvironment(
-  "ARCGIS-API-KEY",
-  defaultValue: "YOUR KEY HERE",
+  "APIKEY",
```
Comment (JulianBissekkou) [inline example/lib/main.dart:13]: Any reason you changed this env variable name?

File: example/lib/main.dart [hunk 6]
```diff
@@ -650,6 +656,12 @@ class _ExampleMapState extends State<ExampleMap> {
                     ],
                   ),
                 ),
+                ElevatedButton(
+                  onPressed: () => _makePolylineVisible(
+                    points: [_firstPinCoordinates, _secondPinCoordinates],
+                  ),
+                  child: const Text('Make polyline visible'),
```
Comment (JulianBissekkou) [inline example/lib/main.dart:663]:
```suggestion
                  child: const Text('Zoom to polyline'),
```

**Review by JulianBissekkou — CHANGES_REQUESTED** [review]

**Review by MatthaiosSait — APPROVED** [review]
Solid 👍

**Comment by passsy** [comment]
Manually merged it into `main`

**Pull request merged**